src/test/java/com/google/sps/ImageTextDectectorTest.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test for ImageTextDectector class. */
+@RunWith(JUnit4.class)
+public final class ImageTextDectectorTest {
+
+  private static final byte[] NULL_IMAGE_BYTES = new byte[0];
+  private static final byte[] IMAGE_BYTES = new byte[1];
+
+  private FakeTextDetectionAPIImpl fakeTextDetectionAPIImpl;
+
+  private ImageTextDectector imageTextDectector;
+
+  @Before
+  public void setUp() {
+    fakeTextDetectionAPIImpl = new FakeTextDetectionAPIImpl();
+  }
+
+  /**
+   * Set text detection result/exception, mocking Cloud Vision API, and initialize ImageTextDectector's
+   * object.
+   */
+  private void initImageTextDectector(List<ShoppingListTextEntry> shoppingListTextEntries) {
+    fakeTextDetectionAPIImpl.setReturnValue(shoppingListTextEntries);
+    imageTextDectector = new ImageTextDectector(fakeTextDetectionAPIImpl);
+  }
+
+  /** Negative test for empty image */
+  @Test
+  public void noBytesImage() throws Exception {
+    String exceptionMessage = "byte array is empty";
+    fakeTextDetectionAPIImpl.setException(new PhotoDetectionException(exceptionMessage));
+    imageTextDectector = new ImageTextDectector(fakeTextDetectionAPIImpl);
+
+    Exception exception =
+        Assertions.assertThrows(
+            PhotoDetectionException.class,
+            () -> {
+              imageTextDectector.extractShoppingList(NULL_IMAGE_BYTES);
+            });
+
+    String expectedMessage = exceptionMessage;
+    String actualMessage = exception.getMessage();
+    Assertions.assertTrue(actualMessage.equals(expectedMessage));
+  }
+
+  /** Negative test for no text */
+  @Test
+  public void noText() throws Exception {
+    List<ShoppingListTextEntry> shoppingListTextEntries= new ArrayList<>();
+
+    initImageTextDectector(shoppingListTextEntries);
+
+    Exception exception =
+        Assertions.assertThrows(
+            PhotoDetectionException.class,
+            () -> {
+              imageTextDectector.extractShoppingList(IMAGE_BYTES);
+            });
+
+    String expectedMessage = "Shopping List doesn't contain any text";
+    String actualMessage = exception.getMessage();
+    Assertions.assertTrue(actualMessage.equals(expectedMessage));
+  }
+
+  @Test
+  public void singleWordImage() throws Exception {
+    List<ShoppingListTextEntry> shoppingListTextEntries= new ArrayList<>();
+    shoppingListTextEntries.add(ShoppingListTextEntry.create("Bag", 10));
+
+    initImageTextDectector(shoppingListTextEntries);
+
+    String expectedShoppingQuery = "Bag";
+    String actualShoppingQuery = imageTextDectector.extractShoppingList(IMAGE_BYTES);
+    Assert.assertEquals(expectedShoppingQuery, actualShoppingQuery);
+  }
+
+  @Test
+  public void multiWord_inSingleLine_Image() throws Exception {
```

Comment:
Don't mix camelCase with underscores. I doubt that this is allowed by the style guide. Same below.

Reply:
Oh yes! I was just thinking that it'll make it more readable.
Will change it.